instrumentation/spring/spring-pulsar-1.2/testing/src/main/java/io/opentelemetry/instrumentation/spring/pulsar/v1_0/AbstractSpringPulsarTest.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.pulsar.annotation.PulsarListener;
+import org.springframework.pulsar.core.PulsarTemplate;
+import org.testcontainers.containers.PulsarContainer;
+import org.testcontainers.utility.DockerImageName;
+
+@SuppressWarnings("deprecation") // using deprecated semconv
+public abstract class AbstractSpringPulsarTest {
+
+  @RegisterExtension
+  protected static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  static final DockerImageName DEFAULT_IMAGE_NAME =
+      DockerImageName.parse("apachepulsar/pulsar:4.0.2");
+  static PulsarContainer pulsarContainer;
+  static ConfigurableApplicationContext applicationContext;
+  static PulsarTemplate<String> pulsarTemplate;
+  static PulsarClient client;
+  static CountDownLatch latch = new CountDownLatch(1);
+  static final String OTEL_SUBSCRIPTION = "otel-subscription";
+  protected static String brokerHost;
+  protected static int brokerPort;
+  protected static final String OTEL_TOPIC = "persistent://public/default/otel-topic";
+
+  @BeforeAll
+  @SuppressWarnings("unchecked")
+  static void setUp() throws PulsarClientException {
+    pulsarContainer =
+        new PulsarContainer(DEFAULT_IMAGE_NAME)
+            .withEnv("PULSAR_MEM", "-Xmx128m")
+            .withStartupTimeout(Duration.ofMinutes(2));
+    pulsarContainer.start();
+    brokerHost = pulsarContainer.getHost();
+    brokerPort = pulsarContainer.getMappedPort(6650);
+
+    SpringApplication app = new SpringApplication(ConsumerConfig.class);
+    Map<String, Object> props = new HashMap<>();
+    props.put("spring.main.web-application-type", "none");
+    props.put("spring.pulsar.client.service-url", pulsarContainer.getPulsarBrokerUrl());
+    props.put("spring.pulsar.consumer.subscription.initial-position", "earliest");
+    app.setDefaultProperties(props);
+    applicationContext = app.run();
+    pulsarTemplate = applicationContext.getBean(PulsarTemplate.class);
+
+    client = PulsarClient.builder().serviceUrl(pulsarContainer.getPulsarBrokerUrl()).build();
+  }
+
+  @Test
+  void testSpringPulsar() throws InterruptedException {
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          pulsarTemplate.send(OTEL_TOPIC, "test");
+        });
+    latch.await(10, TimeUnit.SECONDS);
+    assertSpringPulsar();
+  }
+
+  @AfterAll
+  static void teardown() {
+    if (applicationContext != null) {
+      applicationContext.close();
+    }
+    if (pulsarContainer != null) {
+      pulsarContainer.stop();
+    }
+  }
+
+  protected abstract void assertSpringPulsar();
+
+  static final AttributeKey<String> MESSAGE_TYPE =
+      AttributeKey.stringKey("messaging.pulsar.message.type");
+
+  protected List<AttributeAssertion> publishAttributes() {
+    return Arrays.asList(
```

Comment:
We'd use static import when it's used several time in a class. For detail, you can refer to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/style-guideline.md#static-imports